MigrationManager.php

```diff
@@ -106,188 +102,164 @@ public function migrate($desired_version = null)
+     * Warning: This function works on the assumption that only one class is defined in the
      * migration script!
-     * @param filepath
+     * @param string $filepath
+     * @return string
+     * @throws Exception
      */
-    private function include_file_and_get_class_name($filepath)
+    private function include_file_and_get_class_name(string $filepath): string
     {
         $existingClasses = get_declared_classes();
         require_once($filepath);
         $afterClasses = get_declared_classes();
         $new_classes = array_diff($afterClasses, $existingClasses);
-        
-        if (count($new_classes) == 0)
-        {
+
+        if (count($new_classes) == 0) {
             $errMsg = 'Migration error: Could not find new class from including migration script' .
-                      '. This could be caused by having duplicate class names, or having already ' .
-                      'included the migration script.';
-            
-            throw new \Exception($errMsg);
-        }
-        elseif (count($new_classes) > 1) 
-        {
+                '. This could be caused by having duplicate class names, or having already ' .
+                'included the migration script.';
+
+            throw new Exception($errMsg);
+        } elseif (count($new_classes) > 1) {
             $errMsg = 'Migration error: Found more than 1 class defined in the migration script ' .
-                       '[' . $filepath . ']';
-            
-            throw new \Exception('Migration error:');
+                '[' . $filepath . ']';
+
+            throw new Exception($errMsg);
         }
-        
+
         # newClasses array keeps its keys, so the first element is not at 0 at this point
         $new_classes = array_values($new_classes);
         return $new_classes[0];
     }
-    
 
-        
     /**
-     * Function responsible for deciphering the 'version' from a filename. This is a function 
+     * Function responsible for deciphering the 'version' from a filename. This is a function
      * because we may wish to change it easily.
      * @param string $filename - the name of the file (not full path) that is a migration class.
      * @return int $version - the version the file represents.
      */
-    private static function get_file_version($filename) 
+    private static function get_file_version(string $filename): int
     {
-        $version = intval($filename);
-        return $version;
+        return intval($filename);
     }
-    
-    
+
     /**
      * Inserts the specified version number into the database.
      * @param int $version - the new version of the database.
      * @return void.
+     * @throws Exception
      */
-    private function insert_db_version($version)
+    private function insert_db_version(int $version): void
     {
-        $query = 
-            "REPLACE INTO `" . self::MIGRATIONS_TABLE . "` " . 
+        $query =
+            "REPLACE INTO `$this->m_table` " .
```

Comment:
we probably want to perform escaping here now that table is a passed in variable, rather than a hardcoded string that we knew was safe before.